crates/storage_impl/src/redis/cache.rs

```diff
@@ -173,19 +191,38 @@ impl From<CacheKey> for String {
 impl Cache {
     /// With given `time_to_live` and `time_to_idle` creates a moka cache.
     ///
+    /// `name`        : Cache type name to be used as an attribute in metrics
     /// `time_to_live`: Time in seconds before an object is stored in a caching system before it’s deleted
     /// `time_to_idle`: Time in seconds before a `get` or `insert` operation an object is stored in a caching system before it's deleted
     /// `max_capacity`: Max size in MB's that the cache can hold
-    pub fn new(time_to_live: u64, time_to_idle: u64, max_capacity: Option<u64>) -> Self {
+    pub fn new(
+        name: &'static str,
+        time_to_live: u64,
+        time_to_idle: u64,
+        max_capacity: Option<u64>,
+    ) -> Self {
+        // Record the metrics of manual invalidation of cache entry by the application
+        let eviction_listener = move |_, _, cause| {
+            if let RemovalCause::Explicit = cause {
+                metrics::CACHE_MANUAL_INVALIDATION.add(
+                    &metrics::CONTEXT,
+                    1,
+                    &request::add_attributes([("cache_type", name)]),
+                );
```

Comment:
What do you think about incrementing this metric in the `remove()` method, and recording the eviction count in a separate metric (say `IN_MEMORY_CACHE_EVICTION_COUNT`) with the `RemovalCause` as an attribute? That way, we can also have information on the number of items removed due to expiration, values updated (`Replaced`) and due to size constraints. 

They can hopefully help us take informed decisions on the configuration values (TTL, TTI, capacity) for each of the in-memory cache instances.

Reply:
instead of having this logic in `remove()` and having separate metric, can we have a single generic metric for eviction which has `RemovalCause` as an attribute. And this logic resides in eviction listener alone. this should do that job of counting the eviction for all cause?

Reply:
Sure, that should be fine.